src/data_gradients/datasets/detection/voc_format_detection_dataset.py

```diff
@@ -0,0 +1,190 @@
+
+    #### Expected folder structure
+    Any structure including at least one sub-directory for images and one for xml labels. They can be the same.
+
+    Example 1: Separate directories for images and labels
+    ```
+        dataset_root/
+            ├── images/
+            │   ├── train/
+            │   │   ├── 1.jpg
+            │   │   ├── 2.jpg
+            │   │   └── ...
+            │   ├── test/
+            │   │   ├── ...
+            │   └── validation/
+            │       ├── ...
+            └── labels/
+                ├── train/
+                │   ├── 1.xml
+                │   ├── 2.xml
+                │   └── ...
+                ├── test/
+                │   ├── ...
+                └── validation/
+                    ├── ...
+    ```
+
+    Example 2: Same directory for images and labels
+    ```
+        dataset_root/
+            ├── train/
+            │   ├── 1.jpg
+            │   ├── 1.xml
+            │   ├── 2.jpg
+            │   ├── 2.xml
+            │   └── ...
+            └── validation/
+                ├── ...
+    ```
+
+    **Note**: The label file need to be stored in XML format, but the file extension can be different.
+
+    #### Expected label files structure
+    The label files must be structured in XML format, like in the following example:
+
+    ``` xml
+    <annotation>
+        <object>
+            <name>chair</name>
+            <bndbox>
+                <xmin>1</xmin>
+                <ymin>213</ymin>
+                <xmax>263</xmax>
+                <ymax>375</ymax>
+            </bndbox>
+        </object>
+        <object>
+            <name>sofa</name>
+            <bndbox>
+                <xmin>104</xmin>
+                <ymin>151</ymin>
+                <xmax>334</xmax>
+                <ymax>287</ymax>
+            </bndbox>
+        </object>
+    </annotation>
+    ```
+
+
+    #### Instantiation
+    ```
+    dataset_root/
+        ├── images/
+        │   ├── train/
+        │   │   ├── 1.jpg
+        │   │   ├── 2.jpg
+        │   │   └── ...
+        │   ├── test/
+        │   │   ├── ...
+        │   └── validation/
+        │       ├── ...
+        └── labels/
+            ├── train/
+            │   ├── 1.txt
+            │   ├── 2.txt
+            │   └── ...
+            ├── test/
+            │   ├── ...
+            └── validation/
+                ├── ...
+    ```
+
+    ```python
+    from data_gradients.datasets.detection import VOCFormatDetectionDataset
+
+    train_set = VOCFormatDetectionDataset(root_dir="<path/to/dataset_root>", images_dir="images/train", labels_dir="labels/train")
+    val_set = VOCFormatDetectionDataset(root_dir="<path/to/dataset_root>", images_dir="images/validation", labels_dir="labels/validation")
+    ```
+    """
+
+    def __init__(
+        self,
+        root_dir: str,
+        images_dir: str,
+        labels_dir: str,
+        class_names: List[str],
+        config_path: Optional[str],
```

Comment:
docs for config_path?
I think it would be a good idea demonstrating how to use one and show what its contents should look like.

Reply:
First, added some text in [78a5097](https://github.com/Deci-AI/data-gradients/pull/117/commits/78a50976437c0932e76ae9df54811895d0417173)
Then, I added more detailed example in https://github.com/Deci-AI/data-gradients/pull/117/commits/932bd096ec7dba2a1e1e1f3dd585ea4aa48efca5